app/connectors/EnrolmentStoreProxyConnector.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import cats.data.EitherT
+import config.FrontendAppConfig
+import models.Regime
+import models.enrolment.GroupIds
+import models.error.ApiError
+import models.error.ApiError.{EnrolmentExistsError, MalformedError}
+import models.subscription.response.SubscriptionID
+import org.slf4j.LoggerFactory
+import play.api.http.Status.NO_CONTENT
+import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
+import uk.gov.hmrc.http.HttpReads.is2xx
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class EnrolmentStoreProxyConnector @Inject() (val config: FrontendAppConfig, val http: HttpClient) {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  def enrolmentStatus(regime: Regime, subscriptionID: SubscriptionID)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): EitherT[Future, ApiError, Unit] = {
+    val serviceEnrolmentPattern = s"HMRC-${regime.toUpperCase}-ORG~${regime.toUpperCase}ID~${subscriptionID.subscriptionID}"
+    val submissionUrl           = s"${config.enrolmentStoreProxyUrl}/enrolment-store/enrolments/$serviceEnrolmentPattern/groups"
+    EitherT {
+      http
+        .GET[HttpResponse](
+          submissionUrl
+        )(rds = readRaw, hc = hc, ec = ec)
+        .map {
+          case response if response.status == NO_CONTENT => Right(())
+          case response if is2xx(response.status) =>
+            response.json
+              .asOpt[GroupIds]
+              .map(
+                groupIds => Left(EnrolmentExistsError(groupIds))
```

Comment:
@shaileshp0110 The GroupIds can have principalGroupIds and delegatedGroupIds as empty seq still can return object GroupIds, How can we be sure of EnrolmentExistsError error to be correct

Reply:
@ashwinirudrappa  Yes you are right in that case it will fail   let me fix that